qa/tasks/cephfs/mount.py

```diff
@@ -680,7 +680,7 @@ def ls(self, path=None):
         if path:
             cmd.append(path)
 
-        ls_text = self.run_shell(cmd).stdout.getvalue().strip()
+        ls_text = six.ensure_str(self.run_shell(cmd).stdout.getvalue().strip())
```

Comment:
i'd suggest stop using `six` from now one. as we don't need to be compatible with py2 anymore in master.

Reply:
instead of ignoring `stdout` and `stderr` parameters passed to `run_shell()` , please pass them to `self.client_remote.run()`.

Reply:
Okay, will try to fix it later. Thanks.

Reply:
i am un-resolving this conversation, as i still think that passing `StringIO` to `self.client_remote.run()` is a better fix. `LocalRemoteProcess` uses `BytesIO` by default, so does `LocalRemote.sh()` and `Remote.run()` and `Remote.sh()`. i don't see a reason to the other way around.

Reply:
Sure, will fix it.
Thanks.

Reply:
I suggest to remove run_shell  method and use remote.sh, run_shell is just a code duplication, check it out, it does not give anything usefull:   
```
def run_shell(self, args, wait=True, stdin=None, check_status=True,
                  omit_sudo=True):
        if isinstance(args, str):
            args = args.split()

        args = ["cd", self.mountpoint, run.Raw('&&'), "sudo"] + args
        return self.client_remote.run(args=args, stdout=StringIO(),
                                      stderr=StringIO(), wait=wait,
                                      stdin=stdin, check_status=check_status,
                                      omit_sudo=omit_sudo)
```
Instead of
```
         ls_text = six.ensure_str(self.run_shell(cmd).stdout.getvalue().strip()) 
```
you might want use this:
```
   ls_text = self.client_remote.sh("sudo ls %s" % path, cwd=self.mountpoint).strip()
```

Reply:
Rishabh have fix the partially fixed it in https://github.com/ceph/ceph/pull/34595, and for the py3 compatible is fixing in https://github.com/ceph/ceph/pull/34781.